### PR TITLE
PYPY: pypy does not implement PyCode_NewWithPosOnlyArgs

### PR DESCRIPTION
**Pull request by mattip**
Since it seems this code path will go away in CPython3.11, I don't really want to implement it only to remove it. Would it be reasonable to fall back to the older code path on PyPy?

### CODE REVIEW

**Comment by da-woods** [comment]
This definitely seems reasonable to me. I don't think the extra information is used for anything hugely important anyway.

File: Cython/Utility/ModuleSetupCode.c [hunk 1]
```diff
@@ -537,7 +537,7 @@ class __Pyx_FakeReference {
         }
         return co;
     }
-#elif PY_VERSION_HEX >= 0x030800B2
+#elif PY_VERSION_HEX >= 0x030800B2 && !defined PYPY_VERSION
```
Comment (scoder) [inline Cython/Utility/ModuleSetupCode.c:540]:
```suggestion
#elif PY_VERSION_HEX >= 0x030800B2 && !defined(PYPY_VERSION)
```

File: Cython/Utility/ModuleSetupCode.c [hunk 2]
```diff
@@ -537,7 +537,7 @@ class __Pyx_FakeReference {
         }
         return co;
     }
-#elif PY_VERSION_HEX >= 0x030800B2
+#elif PY_VERSION_HEX >= 0x030800B2 && !defined(PYPY_VERSION)
```
Comment (scoder) [inline Cython/Utility/ModuleSetupCode.c:540]:
```suggestion
#elif PY_VERSION_HEX >= 0x030800B2 && !CYTHON_COMPILING_IN_PYPY
```

**Comment by mattip** [comment]
Thanks

**Pull request merged**